packages/schema-type/src/typescript/SchemaType.ts

```diff
@@ -0,0 +1,639 @@
+  [SchemaType.UINT248_ARRAY]: "uint248[]",
+  [SchemaType.UINT256_ARRAY]: "uint256[]",
+
+  [SchemaType.INT8_ARRAY]: "int8[]",
+  [SchemaType.INT16_ARRAY]: "int16[]",
+  [SchemaType.INT24_ARRAY]: "int24[]",
+  [SchemaType.INT32_ARRAY]: "int32[]",
+  [SchemaType.INT40_ARRAY]: "int40[]",
+  [SchemaType.INT48_ARRAY]: "int48[]",
+  [SchemaType.INT56_ARRAY]: "int56[]",
+  [SchemaType.INT64_ARRAY]: "int64[]",
+  [SchemaType.INT72_ARRAY]: "int72[]",
+  [SchemaType.INT80_ARRAY]: "int80[]",
+  [SchemaType.INT88_ARRAY]: "int88[]",
+  [SchemaType.INT96_ARRAY]: "int96[]",
+  [SchemaType.INT104_ARRAY]: "int104[]",
+  [SchemaType.INT112_ARRAY]: "int112[]",
+  [SchemaType.INT120_ARRAY]: "int120[]",
+  [SchemaType.INT128_ARRAY]: "int128[]",
+  [SchemaType.INT136_ARRAY]: "int136[]",
+  [SchemaType.INT144_ARRAY]: "int144[]",
+  [SchemaType.INT152_ARRAY]: "int152[]",
+  [SchemaType.INT160_ARRAY]: "int160[]",
+  [SchemaType.INT168_ARRAY]: "int168[]",
+  [SchemaType.INT176_ARRAY]: "int176[]",
+  [SchemaType.INT184_ARRAY]: "int184[]",
+  [SchemaType.INT192_ARRAY]: "int192[]",
+  [SchemaType.INT200_ARRAY]: "int200[]",
+  [SchemaType.INT208_ARRAY]: "int208[]",
+  [SchemaType.INT216_ARRAY]: "int216[]",
+  [SchemaType.INT224_ARRAY]: "int224[]",
+  [SchemaType.INT232_ARRAY]: "int232[]",
+  [SchemaType.INT240_ARRAY]: "int240[]",
+  [SchemaType.INT248_ARRAY]: "int248[]",
+  [SchemaType.INT256_ARRAY]: "int256[]",
+
+  [SchemaType.BYTES1_ARRAY]: "bytes1[]",
+  [SchemaType.BYTES2_ARRAY]: "bytes2[]",
+  [SchemaType.BYTES3_ARRAY]: "bytes3[]",
+  [SchemaType.BYTES4_ARRAY]: "bytes4[]",
+  [SchemaType.BYTES5_ARRAY]: "bytes5[]",
+  [SchemaType.BYTES6_ARRAY]: "bytes6[]",
+  [SchemaType.BYTES7_ARRAY]: "bytes7[]",
+  [SchemaType.BYTES8_ARRAY]: "bytes8[]",
+  [SchemaType.BYTES9_ARRAY]: "bytes9[]",
+  [SchemaType.BYTES10_ARRAY]: "bytes10[]",
+  [SchemaType.BYTES11_ARRAY]: "bytes11[]",
+  [SchemaType.BYTES12_ARRAY]: "bytes12[]",
+  [SchemaType.BYTES13_ARRAY]: "bytes13[]",
+  [SchemaType.BYTES14_ARRAY]: "bytes14[]",
+  [SchemaType.BYTES15_ARRAY]: "bytes15[]",
+  [SchemaType.BYTES16_ARRAY]: "bytes16[]",
+  [SchemaType.BYTES17_ARRAY]: "bytes17[]",
+  [SchemaType.BYTES18_ARRAY]: "bytes18[]",
+  [SchemaType.BYTES19_ARRAY]: "bytes19[]",
+  [SchemaType.BYTES20_ARRAY]: "bytes20[]",
+  [SchemaType.BYTES21_ARRAY]: "bytes21[]",
+  [SchemaType.BYTES22_ARRAY]: "bytes22[]",
+  [SchemaType.BYTES23_ARRAY]: "bytes23[]",
+  [SchemaType.BYTES24_ARRAY]: "bytes24[]",
+  [SchemaType.BYTES25_ARRAY]: "bytes25[]",
+  [SchemaType.BYTES26_ARRAY]: "bytes26[]",
+  [SchemaType.BYTES27_ARRAY]: "bytes27[]",
+  [SchemaType.BYTES28_ARRAY]: "bytes28[]",
+  [SchemaType.BYTES29_ARRAY]: "bytes29[]",
+  [SchemaType.BYTES30_ARRAY]: "bytes30[]",
+  [SchemaType.BYTES31_ARRAY]: "bytes31[]",
+  [SchemaType.BYTES32_ARRAY]: "bytes32[]",
+
+  [SchemaType.BOOL_ARRAY]: "bool[]",
+  [SchemaType.ADDRESS_ARRAY]: "address[]",
+
+  [SchemaType.BYTES]: "bytes",
+  [SchemaType.STRING]: "string",
+};
+
+export type ArraySchemaType =
```

Comment:
Should we add a `DyamicSchemaType` (ArraySchemaType | bytes | string) just to make it more clear that ArraySchemaType doesn't include all dynamic types?